python/dgl/data/movielens.py

```diff
@@ -0,0 +1,640 @@
+        (
+            train_u_indices,
+            train_v_indices,
+            train_labels,
+        ) = self._generate_pair_value(train_rating_data)
+        (
+            valid_u_indices,
+            valid_v_indices,
+            valid_labels,
+        ) = self._generate_pair_value(valid_rating_data)
+        test_u_indices, test_v_indices, test_labels = self._generate_pair_value(
+            test_rating_data
+        )
+
+        # reindex u and v, v nodes start after u
+        num_user = len(self._global_user_id_map)
+        train_v_indices += num_user
+        valid_v_indices += num_user
+        test_v_indices += num_user
+
+        self.train_rating_pairs = (
+            tensor(train_u_indices, dtype=data_type_dict["int64"]),
+            tensor(train_v_indices, dtype=data_type_dict["int64"]),
+        )
+        self.valid_rating_pairs = (
+            tensor(valid_u_indices, dtype=data_type_dict["int64"]),
+            tensor(valid_v_indices, dtype=data_type_dict["int64"]),
+        )
+        self.test_rating_pairs = (
+            tensor(test_u_indices, dtype=data_type_dict["int64"]),
+            tensor(test_v_indices, dtype=data_type_dict["int64"]),
+        )
+        self.train_rating_values = tensor(train_labels)
+        self.valid_rating_values = tensor(valid_labels)
+        self.test_rating_values = tensor(test_labels)
+        self.info = {
+            "train_rating_pairs": self.train_rating_pairs,
+            "valid_rating_pairs": self.valid_rating_pairs,
+            "test_rating_pairs": self.test_rating_pairs,
+        }
+
+        # build dgl graph object, which is homogeneous and bidirectional and contains only training edges
+        self.train_graph = graph(
+            (self.train_rating_pairs[0], self.train_rating_pairs[1])
+        )
+        self.valid_graph = graph(
+            (self.valid_rating_pairs[0], self.valid_rating_pairs[1])
+        )
+        self.test_graph = graph(
+            (self.test_rating_pairs[0], self.test_rating_pairs[1])
+        )
+        self.train_graph.edata["etype"] = self.train_rating_values
+        self.valid_graph.edata["etype"] = self.valid_rating_values
+        self.test_graph.edata["etype"] = self.test_rating_values
+
+        self.train_graph = add_reverse_edges(self.train_graph, copy_edata=True)
+        self.valid_graph = add_reverse_edges(self.valid_graph, copy_edata=True)
+        self.test_graph = add_reverse_edges(self.test_graph, copy_edata=True)
+
+        self.train_graph.edata.pop('_ID')
+        self.valid_graph.edata.pop('_ID')
+        self.test_graph.edata.pop('_ID')
+
+    def has_cache(self):
+        if os.path.exists(self.graph_path) and os.path.exists(self.user_feat_path) and os.path.exists(self.movie_feat_path):
+            return True
+        return False
+
+    def save(self):
+        save_graphs(
+            self.graph_path,
+            [self.train_graph, self.valid_graph, self.test_graph],
+        )
```

Comment:
I think other stuff like `self.info` can also be saved with `save_graphs`

Reply:
Oh really? I was supposed that save_graphs() can only be used to save dgl graphs and other objects have to be saved by save_info() :)

Reply:
I think so, but you can have a try to verify that.

Reply:
Oh I guess we can store them in `labels` param of save_graphs()?

Reply:
> I think other stuff like `self.info` can also be saved with `save_graphs`.

I've tested and it won't work. save_graphs() requires input to be a list of DGLGraphs. We can't input self.info into `labels` param since it requires to be a dict with Tensor as values.